components/experimental/src/measure/category.rs

```diff
@@ -0,0 +1,111 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use crate::measure::measureunit::MeasureUnit;
+
+pub trait MeasureUnitCategory {
+    fn category() -> Categories;
+}
+
+/// A [`MeasureUnit`] that is related to a specific category.
+///
+/// This is useful for type inference and for ensuring that the correct units are used.
+pub struct CategorizedMeasureUnit<T: MeasureUnitCategory> {
+    pub category: T,
+    pub unit: MeasureUnit,
+}
+
+/// The categories of [`MeasureUnit`]s.
+pub enum Categories {
+    Length,
+    Area,
+    Volume,
+    Mass,
+    // TODO: add more categories in the next PRs.
+}
+
+/// A [`MeasureUnit`] that is related to the length category.
+pub struct LengthMeasureUnit;
```

Comment:
nit: we should have a `categories` module with individual types in them so they don't need longer names

Reply:
done